### ISSUE TITLE
Unable to parse incoming Flexible content when using with Nova Field Dependency

### ISSUE THREAD

**Issue by onemoreahmad** [whitecube/nova-flexible-content#38]
The most awesome package for nova so far! Thanks for the great work!

However, when I use the package with [Nova Field Dependency Container](https://github.com/epartment/nova-dependency-container) or [Nova Radio Buttons](https://github.com/OwenMelbz/nova-radio-field) for example to show/hide the flexible field depends on other field value. I got an error message saying `"Unable to parse incoming Flexible content, data should be an array"` whenever the field is hidden, it works just fine when it's shown.

<img width="757" alt="Screen Shot 2019-07-03 at 11 54 48 PM" src="https://user-images.githubusercontent.com/1885785/60624745-c25a8400-9dee-11e9-86e6-18f8e1e00978.png">

Even though these packages really hide the field and don't send it's content with the request, but still Flexable Content package tries to resolve the field.

As a quick fix, I wrapped syncAndFillGroups function in the `vendor/whitecube/nova-flexible-content/src/Flexible.php ` with an isset() condition to be like:
```

    protected function syncAndFillGroups(NovaRequest $request, $requestAttribute)
    {
        if( isset($request[$requestAttribute])){
            $raw = json_decode($request[$requestAttribute]);

            if(!is_array($raw)) {
                throw new \Exception("Unable to parse incoming Flexible content, data should be an array.");
            }

            $this->groups = collect($raw)->map(function($item, $key) use ($request) {
                $layout = $item->layout;
                $key = $item->key;
                $attributes = (array) $item->attributes;

                $group = $this->findGroup($key) ?? $this->newGroup($layout, $key);

                if(!$group) return;

                $group->fill(ScopedRequest::scopeFrom($request, $attributes, $key));

                return $group;
            });
        }
    }
```

**Comment by voidgraphics** [comment]
Thank you for the nice words 👍 Super glad the package is useful to you.

It's tricky to provide compatibility with every other Nova package out there. My work partner and creator of this package @Nyratas actually wrote [a post about this](https://medium.com/@toonVandenBos/making-laravel-nova-custom-fields-more-developer-friendly-c776b2a8a2e5) a while back. I encourage other nova package developers out there to read it, it aims to make it easier for everyone's code to work well together from the start.

That being said, if simply adding the `isset` check at the start fixes compatibility with these two, I don't see a reason not to do it. Can you confirm that it solves the problem entirely? I do not have a ton of time to test this myself.

**Comment by onemoreahmad** [comment]
Great!
I just read the article, can't agree more.

I believe the packages mentioned before is compatible with other fields, specially **Nova Field Dependency Container**.

Anyway, while I was reading @Nyratas post I found at the end of it he mentioned this https://gist.github.com/Nyratas/c04b85435a5cb92ac4e09ab506537962#file-nova-field-fillattributefromrequest-serializable-object-php 

so, I copied  `if (!$request->exists($requestAttribute)) return;` and put it at the first of `fillAttribute()` function located just before the mentioned `syncAndFillGroups()` function earlier, and everything works just fine. I didn't read the whole code or did any intensive testing, but I can assure you it works just fine with and without other third party field packages.

**So, before it was :** 

```
    protected function fillAttribute(NovaRequest $request, $requestAttribute, $model, $attribute)
    {
        $attribute = $attribute ?? $this->attribute;

        $this->buildGroups($model, $attribute);

        $this->syncAndFillGroups($request, $requestAttribute);

        $this->value = $this->resolver->set($model, $attribute, $this->groups);
    }
```
**After**

```
    protected function fillAttribute(NovaRequest $request, $requestAttribute, $model, $attribute)
    {
        if (!$request->exists($requestAttribute)) return;     
  
        $attribute = $attribute ?? $this->attribute;

        $this->buildGroups($model, $attribute);

        $this->syncAndFillGroups($request, $requestAttribute);

        $this->value = $this->resolver->set($model, $attribute, $this->groups);
    }
```

Thanks

**Comment by voidgraphics** [comment]
That sounds about right! I will push that change later today.

**Comment by voidgraphics** [comment]
Included in [v0.1.7](https://github.com/whitecube/nova-flexible-content/releases/tag/v0.1.7)

**Comment by milewski** [comment]
Uhm.. nice I was having this exactly same problem.. instead I sent a PR to the dependency field container repo fixing it, somewhat their implementation was doing something wrong: https://github.com/epartment/nova-dependency-container/pull/60

but as it never got merged.... I guess the update here will work too!
